### ISSUE TITLE
Bug: typecheck fails

### ISSUE THREAD

**Issue by mrNuTz** [SeasonedSoftware/remix-forms#57]
tsc output:
```
node_modules/remix-forms/src/getFormValues.ts:16:56 - error TS2345: Argument of type 'string | string[] | ParsedQs | ParsedQs[] | undefined' is not assignable to parameter of type 'FormDataEntryValue | null'.
  Type 'undefined' is not assignable to type 'FormDataEntryValue | null'.

16     values[key as keyof z.infer<Schema>] = coerceValue(value, shape)
```

**Comment by gustavoguichard** [comment]
Hey @mrNuTz , this is weird.. it seems to me that your project shouldn't be running `tsc` agains the `node_modules` folder.
Anyway, which is the version of your `tsc` (Output of `tsc --version`)?

**Comment by mrNuTz** [comment]
Hi @gustavoguichard , I needed to import types, because I extended the Form component, in that case the module ist type-checked even when it's excluded in ts-config.

**Comment by gustavoguichard** [comment]
@mrNuTz and what would be your tsc version?
I know remix-forms doesn't typecheck in other than newer versions of TS

**Comment by mrNuTz** [comment]
@gustavoguichard `yarn tsc --version` --> `Version 4.7.4`

`strict` is `true` in `tsconfig.json`, maybe that has an effect.

this fixes it for me:
```diff
--- a/node_modules/remix-forms/src/getFormValues.ts
+++ b/node_modules/remix-forms/src/getFormValues.ts
@@ -13,7 +13,7 @@ export default async function getFormValues<Schema extends SomeZodObject>(
   for (const key in schema.shape) {
     const value = input[key]
     const shape = schema.shape[key]
-    values[key as keyof z.infer<Schema>] = coerceValue(value, shape)
+    values[key as keyof z.infer<Schema>] = coerceValue(value as FormDataEntryValue, shape)
   }
 
   return values
```

**Comment by gustavoguichard** [comment]
Cool, @mrNuTz ! I'll try to replicate the issue in my local environment. It I'm unable I still think this type casting can go in a PR so others don't have to deal with it.

Thank you for showing the solution ;)

**Comment by danielweinmann** [comment]
@mrNuTz, I think we solved this on #93. We're now using TS `~4.5.5` for development.

I'm closing the issue but feel free to reopen it at any time. Thanks for the contribution!
